### PR TITLE
Feat JSON schema resolver

### PR DESCRIPTION
**Pull request by ShivamJoker**
Trying to fix the json schema :|
Coming from #309

### CODE REVIEW

File: json-schema/jsonSchema.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  ): ValidateFunction => {
+    // compile only if necessary. otherwise return the memoized validation function
+    if (!validate || allErrors !== validateAllFieldCriteria) {
+      allErrors = validateAllFieldCriteria;
+      validate = new Ajv({
+        ...options.ajvOptions,
+        async: false,
+        allErrors,
+        validateSchema: true,
+        transpile: undefined,
```
Comment (bluebill1049) [inline json-schema/jsonSchema.ts:90]: those should be `options`.

File: json-schema/jsonSchema.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+      }).compile(validationSchema);
+    }
+
+    return validate;
+  };
+
+  // compile the validator right away so there is an exception if the schema is invalid
+  getValidator(allErrors);
+
```
Comment (bluebill1049) [inline json-schema/jsonSchema.ts:99]: why do we need to invoke this here?

File: json-schema/jsonSchema.ts [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+  return async (data, _, validateAllFieldCriteria) => {
+    const validate = getValidator(validateAllFieldCriteria);
+    const valid = validate(data) as boolean;
```
Comment (bluebill1049) [inline json-schema/jsonSchema.ts:102]: why we need cast `boolean` here?

**Review by bluebill1049 — COMMENTED** [review]
@jorisre would you be able to take a look at this as well?

File: json-schema/jsonSchema.ts (hunk 1 above)
Comment (ShivamJoker) [reply json-schema/jsonSchema.ts:90]:
```js
      validate = new Ajv(options.ajvOptions).compile(validationSchema);
```
You meant something like this ?

**Comment by bluebill1049** [comment]
nvm this PR looks really outdated. I will give it a try myself.

**Comment by ShivamJoker** [comment]
Okay that would be great :)

**Comment by bluebill1049** [comment]
thanks for taking a look at it @ShivamJoker

**Comment by jorisre** [comment]
This PR is very outdated, the resolver isn't compatible with resolver v2

**Pull request closed without merging**